modules/commons/src/main/java/org/apache/synapse/commons/datasource/DatasourceMBeanRepository.java

```diff
@@ -81,13 +81,22 @@ public void removeMBean(String name) {
                 MBEAN_CATEGORY_DATABASE_CONNECTION_POOL, name);
     }
 
+    /**
+     * Unregisters an MBean related to datasources without removing it from the datasource MBean map.
+     *
+     * @param name the name of the datasource Mbean to be unregistered.
+     */
+    private void unregisterMBean(String name) {
+        MBeanRegistrar.getInstance().unRegisterMBean(MBEAN_CATEGORY_DATABASE_CONNECTION_POOL, name);
+    }
+
     public void clear() {
 
         if (!dataSourcesMBeans.isEmpty()) {
             log.info("UnRegistering DBPool MBeans");
             for (DBPoolView dbPoolView : dataSourcesMBeans.values()) {
                 if (dbPoolView != null) {
-                    removeMBean(dbPoolView.getName());
+                    unregisterMBean(dbPoolView.getName());

```

Comment:
Why do we need a separate method when we are using it only in one place (i.e. not reusing)?

Reply:
Modified it to be used in another place :)